packages/flutter/lib/src/material/button.dart

```diff
@@ -412,6 +417,9 @@ class MaterialButton extends StatelessWidget {
       ),
       shape: buttonTheme.shape,
       child: child,
+      outerConstraints: theme.materialTapTargetSize == MaterialTapTargetSize.expanded
```

Comment:
Any time you have a conditional on an enum, please use a switch, so that we make sure that we handle every case. Otherwise, if we ever add a value to the enum, we'll forget to update every place that uses the enum.

Reply:
Fixed here and elsewhere